src/AvaloniaEdit.Demo/MainWindow.xaml.cs

```diff
@@ -129,6 +129,7 @@ private void SyntaxModeCombo_SelectionChanged(object sender, SelectionChangedEve
 
             string scopeName = _registryOptions.GetScopeByLanguageId(language.Id);
 
+            _textMateInstallation.SetGrammar(null);
```

Comment:
To avoid reparse the document with the wrong grammar. In the time lapse between you change the document and you set the new grammar we set the grammar to null to stop the tokenizer.